### ISSUE TITLE
Possibility of exposing VTL compiler?

### ISSUE THREAD

**Issue by vicary** [functionless/functionless#257]
I am a contributor of `serverless-appsync-plugin` and a maintainer of `serverless-webpack`.

Before AppSync natively supports JavaScript runtime, which may or may not happen, your Velocity compiler looks promising for AppSync users who also uses the Serverless Framework.

Have you considered exposing the compiler for programmatic use?

**Comment by sam-goodwin** [comment]
Interesting idea! Do you have an example of what integration might look like? 

I don't have a problem with pulling it out into its own isolated package. It works against our `Integration` interface which may help us with decoupling but we need to know more about your use-case before we can know for sure.

**Comment by vicary** [comment]
I am imagining something similar to `synthesizeFunctions`, where it compiles TypeScript strings into Velocity strings. I should be able to add some grabbing logic for resolver mapping templates before `serverless-appsync-plugin` do its deployments.

https://github.com/functionless/functionless/blob/23c6be4e4af49a15b8a2f759b10ef0a7d0ca24bb/src/appsync.ts#L353-L356

**Comment by sam-goodwin** [comment]
So you need a function you can call, point at a source file and convert into a single VTl file that would be used as the request or response resolver handler. You don't want the compiler to create bother the request and resolved stages - instead you will have separate js/ts files for each stage?

**Comment by vicary** [comment]
Yes, exposing the compile function to VTL is all I need to plug that into my other plugins.
